### PR TITLE
jt-fixed raw json display on profile page

### PR DESCRIPTION
**Pull request by TG6S85**
Revamped the profile page frontend to include driver, admin, and rider status. Additionally, I created 2 more userFixtures to test the functionality.
NOTE: There are comments in the code noting where I envisioned wheelchair status and pronoun features to be presented. These features are 2 integral parts of the profile page contingent on the PR's for pronouns and wheelchair endpoints.
Here is a photo (running on springboot): 
Screen Shot 2023-06-06 at 3 19 12 PM

Storybook link:
http://localhost:6006/?path=/docs/pages-profilepage--default

Edit by P. Conrad: Or maybe this?

* Before:  (no storybook)
* After: https://ucsb-cs156-s23.github.io/proj-gauchoride-s23-5pm-1/prs/37/storybook/?path=/docs/pages-profilepage--default

### CODE REVIEW

**Comment by TG6S85** [comment]
Ok, basically, fixed the profilepage.test file, added 2 more fixtures to currentUserFixtures (one as a driver, one as a rider), and beautified the actual frontend page.

File: frontend/src/main/pages/ProfilePage.js [hunk 1]
```diff
@@ -4,41 +4,55 @@ import RoleBadge from "main/components/Profile/RoleBadge";
...
-                <ReactJson src={currentUser.root} />
-            </Row>
-        </BasicLayout>
-    );
+  if (!currentUser.loggedIn) {
+    return <p>Not logged in.</p>;
+  }
+
+  const { email, pictureUrl, fullName } = currentUser.root.user;
+//   const emailVerified = currentUser.root.user.emailVerified;
```
Comment (pconrad) [inline frontend/src/main/pages/ProfilePage.js:15]: No commented out code in PRs please

File: frontend/src/main/pages/ProfilePage.js [hunk 2]
```diff
@@ -4,41 +4,55 @@ import RoleBadge from "main/components/Profile/RoleBadge";
...
+        <Col md={2}>
+          <img
+            src={pictureUrl}
+            alt="Profile"
+            className="rounded-circle img-fluid profile-picture mb-3 mb-md-0"
+          />
+        </Col>
+        <Col md>
+          <h2>{fullName}</h2>
+          {/* This is supposed to be pronouns */}
```
Comment (pconrad) [inline frontend/src/main/pages/ProfilePage.js:29]: No commented out code in PRs please, unless the intention is to say: "here is where pronouns will go".

But that might be better communicated in a separate Draft Pr than in a PR that's going into main.

File: frontend/src/main/pages/ProfilePage.js [hunk 3]
```diff
@@ -4,41 +4,55 @@ import RoleBadge from "main/components/Profile/RoleBadge";
...
+          {/* <p className="lead text-muted">{currentUser.root.user.emailVerified ? "Yes": ""}</p> */}
```
Comment (pconrad) [inline frontend/src/main/pages/ProfilePage.js:30]: No commented out code in PRs to main.

It's fine to have commented out code in a draft PR, or in work in progress, but by the time code goes to main, you need to commit: is it in, or it is out?

File: frontend/src/main/pages/ProfilePage.js [hunk 4]
```diff
@@ -4,41 +4,55 @@ import RoleBadge from "main/components/Profile/RoleBadge";
...
+          {/* <p className="lead text-muted">{email}</p> */}
```
Comment (pconrad) [inline frontend/src/main/pages/ProfilePage.js:31]: See above.

File: frontend/src/tests/pages/ProfilePage.test.js [hunk 5]
```diff
@@ -45,10 +45,53 @@ describe("ProfilePage tests", () => {
...
+            <QueryClientProvider client={queryClient}>
+                <MemoryRouter>
+                    <ProfilePage />
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+
+        await waitFor( () => expect(getByText("Phillip Conrad")).toBeInTheDocument() );
+        expect(getByTestId("role-badge-user")).toBeInTheDocument();
+        // expect(getByTestId("role-badge-admin")).toBeInTheDocument();
```
Comment (pconrad) [inline frontend/src/tests/pages/ProfilePage.test.js:70]: Please address the commented out code.

File: frontend/src/tests/pages/ProfilePage.test.js [hunk 6]
```diff
@@ -45,10 +45,53 @@ describe("ProfilePage tests", () => {
...
+        // expect(getByTestId("role-badge-member")).toBeInTheDocument();
```
Comment (pconrad) [inline frontend/src/tests/pages/ProfilePage.test.js:71]: see above

File: frontend/src/tests/pages/ProfilePage.test.js [hunk 7]
```diff
@@ -45,10 +45,53 @@ describe("ProfilePage tests", () => {
...
```
Comment (pconrad) [inline frontend/src/tests/pages/ProfilePage.test.js:92]: see above

**Review by pconrad — CHANGES_REQUESTED** [review]
Commented out code needs to be cleaned up

**Comment by pconrad** [comment]
This is a frontend PR, so please follow the guidelines previously announced in this thread:
* https://ucsb-cs156-s23.slack.com/archives/C050779M19P/p1685400785702049

File: frontend/src/main/pages/ProfilePage.js (hunk 1 above)
Comment (TG6S85) [reply frontend/src/main/pages/ProfilePage.js:15]: Noted, has been taken care of

File: frontend/src/main/pages/ProfilePage.js (hunk 2 above)
Comment (TG6S85) [reply frontend/src/main/pages/ProfilePage.js:29]: Noted, was planning on adding the pronoun feature once pronoun PR is merged, will be wary about this. I might make a comment so it's easier for future gen to understand the vision if that is ok.

File: frontend/src/tests/pages/ProfilePage.test.js (hunk 5 above)
Comment (TG6S85) [reply frontend/src/tests/pages/ProfilePage.test.js:70]: Noted and addressed.

**Pull request merged**